Choosing option A, breast cancer data set and it should be a binary problem.

In [7]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, LabelEncoder
from sklearn.decomposition import PCA
from sklearn.metrics import accuracy_score, classification_report
import pandas as pd
import numpy as np



In [ ]:
from sklearn.datasets import load_breast_cancer

bc = load_breast_cancer(as_frame=True)
df = bc.frame

print(df.head())
print(df.shape)
display(df.describe)
display(df.info())

   mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst texture  worst perimeter  worst area  \
0             

<bound method NDFrame.describe of      mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0          17.99         10.38          122.80     1001.0          0.11840   
1          20.57         17.77          132.90     1326.0          0.08474   
2          19.69         21.25          130.00     1203.0          0.10960   
3          11.42         20.38           77.58      386.1          0.14250   
4          20.29         14.34          135.10     1297.0          0.10030   
..           ...           ...             ...        ...              ...   
564        21.56         22.39          142.00     1479.0          0.11100   
565        20.13         28.25          131.20     1261.0          0.09780   
566        16.60         28.08          108.30      858.1          0.08455   
567        20.60         29.33          140.10     1265.0          0.11780   
568         7.76         24.54           47.92      181.0          0.05263   

     mean compactness  mean c

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   mean radius              569 non-null    float64
 1   mean texture             569 non-null    float64
 2   mean perimeter           569 non-null    float64
 3   mean area                569 non-null    float64
 4   mean smoothness          569 non-null    float64
 5   mean compactness         569 non-null    float64
 6   mean concavity           569 non-null    float64
 7   mean concave points      569 non-null    float64
 8   mean symmetry            569 non-null    float64
 9   mean fractal dimension   569 non-null    float64
 10  radius error             569 non-null    float64
 11  texture error            569 non-null    float64
 12  perimeter error          569 non-null    float64
 13  area error               569 non-null    float64
 14  smoothness error         5

None

In [13]:
# Set X and y as features and targets
X, y = bc.data, bc.target

# Split data into a 80/20 training, test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)

# Scale the training and test
scaler = StandardScaler()
X_trained = scaler.fit_transform(X_train)
X_testing = scaler.transform(X_test)


In [16]:
# Use PCA
pca_bin = PCA(n_components=0.95)
X_trained_pca = pca_bin.fit_transform(X_trained)
X_test_pca = pca_bin.transform(X_testing)

print("Number of components kept:", pca_bin.n_components_)

# Train the random forest model
rf_model = RandomForestClassifier(n_estimators=100, random_state=42)
rf_model.fit(X_trained_pca, y_train)

y_prediction = rf_model.predict(X_test_pca)

Number of components kept: 10


In [17]:
# Give test scores with accuracy score/classification report
print("accuracy_score \n", accuracy_score(y_test, y_prediction))

print("classification_report \n", classification_report(y_test, y_prediction))


accuracy_score 
 0.9210526315789473
classification_report 
               precision    recall  f1-score   support

           0       0.88      0.90      0.89        42
           1       0.94      0.93      0.94        72

    accuracy                           0.92       114
   macro avg       0.91      0.92      0.92       114
weighted avg       0.92      0.92      0.92       114



How well did Random Forest perform on your dataset?

I believe that this was a veery good result with an accuracy score of 92%. The breast cancer data set has a lot of features so it's impressive it managed to get a good accuracy.

What did PCA do in your workflow?

PCA reduced the number of features down to the 10 most important ones. This simplifies the complexity a little bit and gives more specific features to focus on. If we're to determine 
what was most imporant in the data set, that PCA was absolutely necessary.

Do you think PCA helped or hurt this model? Explain briefly
I believe that PCA did help this model because with the original 39 features of the data set, it can be hard to tell which features are correlated or even important so PCA is important
in simplifiying that for us. It would also help in the visualizations if we were to create them with the number of features being cut to 10.